examples/reuse/reusable_with_mem.py

```diff
@@ -0,0 +1,71 @@
+import asyncio
+
+from async_lru import alru_cache
+
+import flyte
+
+env_reuse = flyte.TaskEnvironment(
+    name="reusable_with_mem",
+    resources=flyte.Resources(memory="500Mi", cpu=1),
+    reusable=flyte.ReusePolicy(
+        replicas=1,  # Min of 2 replicas are needed to ensure no-starvation of tasks.
+        idle_ttl=300,
+    ),
+    image=flyte.Image.from_debian_base().with_pip_packages("unionai-reuse==0.1.3"),
+)
+
+env = env_reuse.clone_with(
+    name="nonreusable_with_mem",
+    reusable=None,
+    depends_on=[env_reuse],
+)
+
+
+class ValueCalculator:
+    """Encapsulates stateful computation to avoid global variables."""
+
+    def __init__(self, initial_value: int = 10):
+        self.v = initial_value
+
+    @alru_cache(maxsize=1)
+    async def get_value(self, x: int) -> int:
+        """
+        Simulate a long-running computation.
+        """
+        result = x * self.v
+        self.v = self.v + 1
+        return result
+
+
+# Create a singleton instance for the reusable environment
+_calculator = ValueCalculator()
```

Comment:
there will be one of these per actor replica right? so the state will be independent across replicas?

Reply:
that is true, so the comment can be improved